torch/_decomp/decompositions.py

```diff
@@ -1200,3 +1203,23 @@ def stack(tensors: List[Tensor], dim: int = 0) -> Tensor:
         return out.view(result_sizes)
     else:
         return torch.cat(get_stack_inputs(tensors, wrapped_dim), dim)
+
+
+def _squeeze_multiple(self, dims):
+    ndims = self.dim()
+    for idx in range(ndims - 1, -1, -1):
+        if idx in dims:
+            self = self.squeeze(idx)
+    return self
+
+
+@register_decomposition(aten.logsumexp.default)
+@pw_cast_for_int_to_real
+def logsumexp(self: Tensor, dim: List[int], keepdim: bool = False) -> Tensor:
+    if self.numel() == 0:
```

Comment:
This will add a guard that has to compute a product, do we want that?

Reply:
Filed an issue about it: https://github.com/pytorch/pytorch/issues/77232